File: SDKs/flutter/lib/src/ivx_config.dart
class IVXConfig {
  /// Game ID (UUID) for your title on the IntelliVerseX platform.
  ///
  /// Copy it from the developer dashboard, or obtain it by calling
  /// `POST https://msapi.intelli-verse-x.io/api/games/game/info` with your game credentials.
  final String gameId;
  final String nakamaHost;
  final int nakamaPort;
  final String nakamaServerKey;
  final bool useSSL;
  final bool enableAnalytics;
  final bool enableDebugLogs;
  final bool verboseLogging;

  const IVXConfig({
    this.gameId = '',
    this.nakamaHost = '127.0.0.1',
    this.nakamaPort = 7350,
    this.nakamaServerKey = 'defaultkey',
    this.useSSL = false,
    this.enableAnalytics = true,
    this.enableDebugLogs = false,
    this.verboseLogging = false,
  });

  void validate() {
    if (nakamaPort < 1 || nakamaPort > 65535) {
      throw ArgumentError('Invalid port: $nakamaPort. Must be 1-65535.');
    }
    if (nakamaHost.trim().isEmpty) {
      throw ArgumentError('nakamaHost cannot be empty.');
    }
    if (nakamaServerKey.trim().isEmpty) {
      throw ArgumentError('nakamaServerKey cannot be empty.');
    }
  }
}

File: SDKs/javascript/src/IVXConfig.ts
// Copyright (c) 2026 Intelli-verse-X
// MIT License — see LICENSE in the project root.

export interface IVXConfig {
  /**
   * Game ID (UUID) for your title on the IntelliVerseX platform.
   * Copy it from the developer dashboard, or obtain it by calling
   * `POST https://msapi.intelli-verse-x.io/api/games/game/info` with your game credentials.
   */
  gameId?: string;
  nakamaHost?: string;
  nakamaPort?: number;
  nakamaServerKey?: string;
  useSSL?: boolean;
  enableAnalytics?: boolean;
  enableDebugLogs?: boolean;
  verboseLogging?: boolean;
}

export const DEFAULT_CONFIG: Required<IVXConfig> = {
  gameId: '',
  nakamaHost: 'nakama-rest.intelli-verse-x.ai',
  nakamaPort: 443,
  nakamaServerKey: 'defaultkey',
  useSSL: true,
  enableAnalytics: true,
  enableDebugLogs: false,
  verboseLogging: false,
};

export function validateConfig(config: IVXConfig): void {
  const gameId = config.gameId ?? '';
  if (gameId.trim() === '') {
    console.warn(
      '[IntelliVerseX] gameId is not set. Set gameId to your Game UUID from the IntelliVerseX dashboard, or obtain it via POST https://msapi.intelli-verse-x.io/api/games/game/info.',
    );
  }
  if (config.nakamaPort !== undefined && (config.nakamaPort < 1 || config.nakamaPort > 65535)) {
    throw new Error(`Invalid port: ${config.nakamaPort}. Must be 1-65535.`);
  }
  if (config.nakamaHost !== undefined && config.nakamaHost.trim() === '') {
    throw new Error('nakamaHost cannot be empty.');
  }
  if (config.nakamaServerKey !== undefined && config.nakamaServerKey.trim() === '') {
    throw new Error('nakamaServerKey cannot be empty.');
  }
}
